src/main/java/net/jodah/typetools/TypeResolver.java

```diff
@@ -250,7 +239,7 @@ public static Type resolveGenericType(Class<?> type, Type subType) {
     Map<TypeVariable<?>, Type> map = ref != null ? ref.get() : null;
 
     if (map == null) {
-      map = new HashMap<TypeVariable<?>, Type>();
+      map = new HashMap<>();
```

Comment:
TypeTools compilation is currently targeted at 1.7, so I don't think we can use the diamond operator.